engine/src/exec-interface-object.cpp

```diff
@@ -3107,7 +3107,10 @@ void MCObject::GetOpaque(MCExecContext& ctxt, bool& r_setting)
 void MCObject::SetOpaque(MCExecContext& ctxt, bool setting)
 {
 	if (changeflag(setting, F_OPAQUE))
+	{
+		sync_mfocus();
```

Comment:
So looking at the engine source, hittest() appears to only change (based on the state of `opaque`) for MCGroup and MCGraphic. If `sync_mfocus` doesn't actually do anything significant if hittesting hasn't changed, then this is okay - otherwise we need to consider only doing this for controls which do affect hittesting (e.g. by override SetOpaque in MCGroup, like it is in MCGraphic).

Reply:
Oh! Good point :-) `sync_mfocus` does do quite a bit yeah, I would say definitely worth overriding SetOpaque for groups